controllers/managedocs_controller.go

```diff
@@ -255,6 +265,95 @@ func (r *ManagedOCSReconciler) SetupWithManager(mgr ctrl.Manager, ctrlOptions *c
 		},
 	)
 
+	var options client.Options
+	var err error
+	var unrestrictedClient client.Client
+
+	options.Scheme = runtime.NewScheme()
+	utilruntime.Must(apiextenstionsv1.AddToScheme(options.Scheme))
+
+	unrestrictedClient, err = client.New(config.GetConfigOrDie(), options)
+	if err != nil {
+		r.Log.Error(err, "error creating client")
+		os.Exit(1)
+	}
+
+	egreeNetworkPolicyCRD := &unstructured.Unstructured{}
+	egreeNetworkPolicyCRD.SetGroupVersionKind(schema.GroupVersionKind{
+		Group:   "apiextensions.k8s.io",
+		Kind:    "CustomResourceDefinition",
+		Version: "v1",
+	})
+	ctx := context.TODO()
+	defer ctx.Done()
+	if err = unrestrictedClient.Get(context.TODO(), client.ObjectKey{Name: "egressnetworkpolicies.network.openshift.io", Namespace: ""}, egreeNetworkPolicyCRD); err != nil {
```

Comment:
```suggestion
	var options client.Options
	var err error

	options.Scheme = runtime.NewScheme()
	utilruntime.Must(apiextenstionsv1.AddToScheme(options.Scheme))

	egreeNetworkPolicyCRD := &unstructured.Unstructured{}
	egreeNetworkPolicyCRD.SetGroupVersionKind(schema.GroupVersionKind{
		Group:   "apiextensions.k8s.io",
		Kind:    "CustomResourceDefinition",
		Version: "v1",
	})
	ctx := context.TODO()
	defer ctx.Done()
	if err = r.UnrestrictedClient.Get(context.TODO(), client.ObjectKey{Name: "egressnetworkpolicies.network.openshift.io", Namespace: ""}, egreeNetworkPolicyCRD); err != nil {
```
- did you try above, type already has the field to use unrestricted client?

Reply:
Yes, it works. I have updated the PR

Reply:
done

Reply:
ack